lib/config/config-loader.js

```diff
@@ -499,11 +499,41 @@ class ConfigLoader {
             debug(`Loading config file ${configFilePath}`);
             const fileConfig = await loadConfigFile(configFilePath);
 
-            if (Array.isArray(fileConfig)) {
-                configs.push(...fileConfig);
-            } else {
-                configs.push(fileConfig);
+            /*
+             * It's possible that a config file could be empty or else
+             * have an empty object or array. In this case, we want to
+             * warn the user that they have an empty config.
+             *
+             * An empty CommonJS file exports an empty object while
+             * an empty ESM file exports undefined.
+             */
+
+            let emptyConfig = typeof fileConfig === "undefined";
+
+            debug(`Config file ${configFilePath} is ${emptyConfig ? "empty" : "not empty"}`);
+
+            if (fileConfig) {
+                if (Array.isArray(fileConfig)) {
+                    if (fileConfig.length === 0) {
+                        debug(`Config file ${configFilePath} is an empty array`);
+                        emptyConfig = true;
+                    } else {
+                        configs.push(...fileConfig);
+                    }
+                } else {
+                    if (Object.keys(fileConfig).length === 0) {
+                        debug(`Config file ${configFilePath} is an empty object`);
+                        emptyConfig = true;
+                    } else {
+                        configs.push(fileConfig);
+                    }
+                }
             }
+
+            if (emptyConfig) {
+                globalThis.process?.emitWarning?.(`Running ESLint with an empty config (from ${configFilePath}). Please double-check that this is what you want.`, "ESLintEmptyConfigWarning");
```

Comment:
```suggestion
                globalThis.process?.emitWarning?.(`Running ESLint with an empty config (from ${configFilePath}). Please double-check that this is what you want. If you want to run ESLint with an empty config, export [{}] to remove this warning.`, "ESLintEmptyConfigWarning");
```

A suggestion on how to remove the warning if empty config is intentional.